### PR TITLE
Use makefiles, Add test runner + codecov

### PR DESCRIPTION
**Pull request by jakebolam**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,43 @@
+# so we can use `eslint` without ./node_modules/.bin/eslint
+SHELL := /bin/bash
+export PATH := $(shell npm bin):$(PATH)
+
+ARTIFACT_DIR?=artifacts
+TEST_REPORTS_DIR?=$(ARTIFACT_DIR)/reports
+BUILD_DIR?=$(ARTIFACT_DIR)/build
+
+IGNORE = test.js,js.snap,.md,__tests__,mockdata,__mocks__
+BABEL_FLAGS = src --out-dir $(BUILD_DIR) --copy-files --ignore $(IGNORE)
```
Comment (jakebolam) [inline Makefile]: Setup this up for babel, will do the babel stuff in another PR

Comment (francoiscampbell) [reply Makefile]: FYI, since we'll be using webpack, we won't need to manually call babel

Comment (jakebolam) [reply Makefile]: Didn't know we were going down this path, can we remove this on that PR?

File: package.json [hunk 2]
```diff
@@ -10,6 +10,17 @@
   "scripts": {
     "precommit": "lint-staged"
   },
+  "jest": {
+    "testPathIgnorePatterns": [
+      "/lib/",
```
Comment (francoiscampbell) [inline package.json]: we don't have `/lib/` do we?

File: Makefile (hunk 1 above)
Comment (francoiscampbell) [reply Makefile]: yup, i'll take care of it
